parser.js

```diff
@@ -0,0 +1,84 @@
+const getMarkdown = (str, re, tag) => {
+	const result = str.match(re);
+
+	if (result && result.length) {
+		return `<${tag}>${result[0]}</${tag}>`;
+	}
+
+	return str;
+};
+
+const rules = [
+	{
+		name: 'Heading level 1',
+		tag: 'h1',
+		regex: /(?<=^# )(.*)/,
+		parse: getMarkdown,
+	},
+	{
+		name: 'Heading level 2',
+		tag: 'h2',
+		regex: /(?<=^## )(.*)/,
+		parse: getMarkdown,
+	},
+	{
+		name: 'Heading level 3',
+		tag: 'h3',
+		regex: /(?<=^### )(.*)/,
+		parse: getMarkdown,
+	},
+	{
+		name: 'Heading level 4',
+		tag: 'h4',
+		regex: /(?<=^#### )(.*)/,
+		parse: getMarkdown,
+	},
+	{
+		name: 'Item',
+		tag: 'li',
+		regex: /(?<=^- )(.*)/,
+		parse: getMarkdown,
+	},
+];
+
+function parser(text) {
```

Comment:
I also suggest that you divide this parser function into smaller functions with a single responsibility (SRP). 

This will make the code more readable, and will help making it easier to change, because focusing on a single smaller function reduces the chances that you will break something else.